### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by kpp**
Mostly removing redundant `&` from references. This PR reduces the number of clippy warnings.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -80,7 +80,7 @@ libp2p-ping = { version = "0.31.0", path = "protocols/ping", optional = true }
 libp2p-plaintext = { version = "0.30.0", path = "transports/plaintext", optional = true }
 libp2p-pnet = { version = "0.22.0", path = "transports/pnet", optional = true }
 libp2p-relay = { version = "0.4.0", path = "protocols/relay", optional = true }
-libp2p-rendezvous = { version = "0.1.0", path = "protocols/rendezvous", optional = true }
+libp2p-rendezvous = { version = "0.1.1", path = "protocols/rendezvous", optional = true }
```
Comment (mxinden) [inline Cargo.toml:83]:
```suggestion
libp2p-rendezvous = { version = "0.1.1", path = "protocols/rendezvous", optional = true }
```
```suggestion
libp2p-rendezvous = { version = "0.1.0", path = "protocols/rendezvous", optional = true }
```

File: protocols/rendezvous/Cargo.toml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 name = "libp2p-rendezvous"
 edition = "2018"
 description = "Rendezvous protocol for libp2p"
-version = "0.1.0"
+version = "0.1.1"
```
Comment (mxinden) [inline protocols/rendezvous/Cargo.toml:5]:
```suggestion
version = "0.1.0"
```

**Comment by mxinden** [comment]
Pull request fails in rustfmt stage. Could you push a fix @kpp?

**Comment by kpp** [comment]
Yes

**Comment by mxinden** [comment]
For the future, please don't force push to a reviewed pull request @kpp. There is no easy way for me to see the diff, thus I have to review the whole pull request once more. Given that we squash all pull requests, many commits do not mess up our commit history on `master`. Thanks :)

**Comment by kpp** [comment]
Ah I see. Sure =) Sorry for that :D

**Review by mxinden — APPROVED** [review]
Thanks for the help!

**Pull request merged**